reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -72,6 +94,7 @@ static Context of(Object key, Object value) {
 	 */
 	static Context of(Object key1, Object value1,
 			Object key2, Object value2) {
+		checkDuplicateKeys(key1, key2);
```

Comment:
AFAIR this creates an allocation (an array of varargs), while the implementation is as simple as `if (key1.equals(key2))`.
Maybe we shouldn't use the generic check method here?

Reply:
good point, originally intended to limit the impact for the 2 k/v case (which is likely one of the most used along with 1 k/v and `empty()` cases) but moving that special case from `checkDuplicateKey` to the main method would save additional allocation.